### PR TITLE
Add transport exception check for delete message

### PR DESCRIPTION
**Pull request by howard-lin-zocdoc**
@chris-smith-zocdoc

### CODE REVIEW

File: src/Twilio.Api/Messages.cs [hunk 1]
```diff
@@ -34,9 +34,14 @@ public virtual DeleteStatus DeleteMessage(string messageSid)
...
-			return response.StatusCode == System.Net.HttpStatusCode.NoContent ? DeleteStatus.Success : DeleteStatus.Failed;
-		}
+            var response = Execute(request);
+            if (response.ResponseStatus != ResponseStatus.Completed ||
+                response.StatusCode != System.Net.HttpStatusCode.NoContent)
+            {
+                throw new TransportException(response);
+            }
+            return DeleteStatus.Success;
+        }
```
Comment (chris-smith-zocdoc) [inline src/Twilio.Api/Messages.cs:44]: tabs vs spaces?

File: src/Twilio.Api/TransportException.cs [hunk 2]
```diff
@@ -5,11 +5,11 @@
 
 namespace Twilio
 {
-    public class TransportException<T> : Exception
```
Comment (howard-lin-zocdoc) [inline src/Twilio.Api/TransportException.cs:8]: @chris-smith-zocdoc is the type information here useful? Does it get print out as part of the Exception? 
`DeleteMessage()` is using a none generic version of `Execute`.

File: src/Twilio.Api/TransportException.cs [hunk 3]
```diff
@@ -5,11 +5,11 @@
 
 namespace Twilio
 {
-    public class TransportException<T> : Exception
+    public class TransportException : Exception
     {
-        public IRestResponse<T> Response { get; private set; }
+        public IRestResponse Response { get; private set; }
 
-        public TransportException(IRestResponse<T> response)
```
Comment (chris-smith-zocdoc) [inline src/Twilio.Api/TransportException.cs:12]: why do we need to remove the generic type?

File: src/Twilio.Api/TransportException.cs (hunk 2 above)
Comment (chris-smith-zocdoc) [inline src/Twilio.Api/TransportException.cs:8]: We catch `TransportException<Twilio.Message>` from within our code, so you would need to update that also.  I'm not sure if we need the type information though. Checkout our usages

Comment (howard-lin-zocdoc) [inline src/Twilio.Api/TransportException.cs:8]: Nice good to know, I will change our usage to catch differently.

Comment (howard-lin-zocdoc) [inline src/Twilio.Api/TransportException.cs:8]: We dont print out the generic type in any of the logging. We should be safe to remove this.

**Pull request closed without merging**